search/search_repository_whitebox_test.go

```diff
@@ -45,189 +52,242 @@ func (s *searchRepositoryWhiteboxTest) SetupTest() {
 }
 
 type SearchTestDescriptor struct {
-	wi             workitem.WorkItem
+	fields         map[string]interface{}
 	searchString   string
 	minimumResults int
 }
 
-func (s *searchRepositoryWhiteboxTest) TestSearchByText() {
-	wir := workitem.NewWorkItemRepository(s.DB)
-
-	testDataSet := []SearchTestDescriptor{
-		{
-			wi: workitem.WorkItem{
-				Fields: map[string]interface{}{
+func (s *searchRepositoryWhiteboxTest) TestSearch() {
+	// create 2 work items, the second one having the number of the first one in its title
+	baseFxt, err := tf.NewFixture(s.DB, tf.Identities(2, tf.SetIdentityUsernames([]string{"creator", "assignee"})),
+		tf.Spaces(1, func(fxt *tf.TestFixture, idx int) error {
+			fxt.Spaces[idx].OwnerId = fxt.Identities[0].ID
```

Comment:
No need to write this down explicitly as a space owner is automatically set to the first identity. Look at `makeSpaces()`.

That being said, the whole fixture (6 lines) can be compressed to just:

```go
tf.NewTestFixture(s.DB, s.T(), tf.Identities(2), tf.Spaces(1), tf.WorkItemTypes(1))
```

Below you're using `baseFxt.Identities[0].ID.String(),` *without specifying an identities username*. Therefore you might as well just put my fixture recipe (`tf.Identities(2), tf.Spaces(1), tf.WorkItemTypes(1)`) from above into every test fixture of every sub-test below. Then you can get rid of these three lines that adjusted the space and the creator:

```go
fxt.WorkItems[idx].SpaceID = baseFxt.Spaces[0].ID
fxt.WorkItems[idx].Type = baseFxt.WorkItemTypes[0].ID
//...
workitem.SystemCreator:     baseFxt.Identities[0].ID.String(),
```

Please note, that it would help me a lot if you do this because when space templates come in and you create two fixtures that use each other, things can become complicated to adjust for me.

If the sate of a workitem is also irrelevant, you can simply ignore it and instead of specifying the whole `map` of Fields you might as well only set those keys you want. Know, that we [pre-fill](https://github.com/fabric8-services/fabric8-wit/blob/5297408457360d776f502a75d6052a1f62d8bbf0/test/testfixture/make_functions.go#L307) the state to `workitem.SystemStateNew` and the creator field to the first identity known to the fixture.

Reply:
ok, I simplified the test here. Please let mw know if it's good for you.

Reply:
Now that is reaaaaaly clean now. Let me have a look at the rest of this PR.